assembl/views/api/post.py

```diff
@@ -313,16 +373,68 @@ def get_posts(request):
         posts = posts.order_by(Content.body_text_index.score_name.desc())
     else:
         posts = posts.order_by(Content.id)
-    print str(posts)
+    # print str(posts)
 
     no_of_posts = 0
     no_of_posts_viewed_by_user = 0
 
+
+    def post_and_its_whole_line_of_descent_are_deleted(post):
+        """
+        :returns: True if post and its whole line of descent (direct and indirect answers) are deleted, False otherwise.
+        :rtype: bool
+        :param post: The post you want to analyse
+        :type post: assembl.models.Post
+
+        """
+        if not post.is_tombstone:
+            return False
+        children = post.children
+        if len(children) == 0:
+            return True
+        for child in children:
+            if not post_and_its_whole_line_of_descent_are_deleted(child):
+                return False
+        return True
+
+    def post_or_one_of_line_of_descent_is_deleted(post):
+        """
+        :returns: True if post or one of its line of descent (direct and indirect answers) is deleted, False otherwise.
+        :rtype: bool
+        :param post: The post you want to analyse
+        :type post: assembl.models.Post
+        
+        """
+        if post.is_tombstone:
+            return True
+        children = post.children
+        if len(children) == 0:
```

Comment:
I think this test is useless. When they are no children, you won't enter the loop and consequently False will be returned.